tests/app/questionnaire/test_router.py

```diff
@@ -506,6 +506,101 @@ def test_section_summary_on_completion_false(self):
+                "sixth-number-block",
+                "currency-total-playback-skipped-fourth",
+            ],
+            section_id="default-section",
+        )
+
+        next_location_url = self.router.get_next_location_url(
+            current_location,
+            routing_path,
+            return_to_answer_id="first-number-answer",
+            return_to="calculated-summary",
+            return_to_block_id="currency-total-playback-skipped-fourth",
+        )
+        expected_location = Location(
+            section_id="default-section",
+            block_id="currency-total-playback-skipped-fourth",
+        )
+
+        expected_location_url = url_for(
+            "questionnaire.block",
+            list_item_id=expected_location.list_item_id,
+            block_id=expected_location.block_id,
+            _anchor="first-number-answer",
+        )
+
+        assert expected_location_url == next_location_url
+
+    @pytest.mark.usefixtures("app")
+    @pytest.mark.parametrize(
+        "return_to_block_id, expected_url",
+        [
+            (
+                "non-valid-block",
+                "/questionnaire/sixth-number-block/?return_to=calculated-summary&return_to_block_id=non-valid-block",
+            ),
+            (None, "/questionnaire/sixth-number-block/?return_to=calculated-summary"),
+        ],
+    )
+    def test_return_to_calculated_summary_invalid_return_to_block_id(
+        self, return_to_block_id, expected_url
+    ):
+        self.schema = load_schema_from_name("test_calculated_summary")
+
+        current_location = Location(
+            section_id="default-section", block_id="fifth-number-block"
+        )
+
+        routing_path = RoutingPath(
+            ["fifth-number-block", "sixth-number-block"],
+            section_id="default-section",
+        )
+        next_location_url = self.router.get_next_location_url(
+            current_location,
+            routing_path,
+            return_to="calculated-summary",
+            return_to_block_id=return_to_block_id,
+        )
+
+        assert expected_url == next_location_url
+
+    @pytest.mark.usefixtures("app")
+    def test_return_to_calculated_summary_return_to_block_id_not_on_path(self):
+        self.schema = load_schema_from_name("test_calculated_summary")
+
+        current_location = Location(
+            section_id="default-section", block_id="fifth-number-block"
+        )
+
+        routing_path = RoutingPath(
+            ["fifth-number-block", "sixth-number-block"],
+            section_id="default-section",
+        )
+        next_location_url = self.router.get_next_location_url(
+            current_location,
+            routing_path,
+            return_to="calculated-summary",
+            return_to_block_id="fourth-number-block",
+        )
+
+        assert (
+            "/questionnaire/sixth-number-block/?return_to=calculated-summary&return_to_block_id=fourth-number-block"
+            == next_location_url
+        )
```

Comment:
This warrants a comment to explain why the same query string is passed along. i.e it may be on the path in future once incomplete questions are answered.